src/sider/ftrl/saksbehandling/stegKomponenter/vurderingBestemmelse/vurderingBestemmelse.tsx

```diff
@@ -97,10 +98,19 @@ export const VurderingBestemmelse = ({ bekreft, tilbake, aktivtSteg, oppdaterSta
     const valgtBestemmelseMedVilkårOgBegrunnelser = støttedeBestemmelser.find(
       (element) => element.bestemmelse === valgtBestemmelse
     );
+
     const alleVilkårHarSvarJaOgValgtBegrunnelse = valgtBestemmelseMedVilkårOgBegrunnelser?.vilkårOgBegrunnelser.every(
-      (element) =>
-        valgteVilkår.get(element.vilkår) === SANN &&
-        (Utils._isEmpty(element.muligeBegrunnelser) ? true : valgteBegrunnelser.get(`${element.vilkår}_begrunnelser`))
+      (element) => {
+        const valgtBegrunnelseForVilkår = valgteBegrunnelser.get(`${element.vilkår}_begrunnelser`);
+        const vilgårBegrunnelser =
+          valgtBegrunnelseForVilkår?.begrunnelseKode === "ANNEN_GRUNN" ? valgtBegrunnelseForVilkår : undefined;
+        const erGyldigAnnenGrunn =
+          !vilgårBegrunnelser || harStrengInnhold(vilgårBegrunnelser?.begrunnelseFritekst ?? "");
+        const harVilkår = valgteVilkår.get(element.vilkår) === SANN;
+        const harGyldigValgtBegrunnelse = Utils._isEmpty(element.muligeBegrunnelser) ? true : valgtBegrunnelseForVilkår;
+
+        return harVilkår && harGyldigValgtBegrunnelse && erGyldigAnnenGrunn;
```

Comment:
```suggestion
      (element) => {
        const harVilkår = valgteVilkår.get(element.vilkår) === SANN;
        const valgtBegrunnelseForVilkår = valgteBegrunnelser.get(`${element.vilkår}_begrunnelser`);
        const begrunnelseInneholderFritekst = KV.termFraNestedKTObject(
          begrunnelseKodeverk,
          valgtBegrunnelseForVilkår?.begrunnelseKode
        )?.includes("(fritekst)");

        if (Utils._isEmpty(element.muligeBegrunnelser)) {
          return harVilkår;
        }

        if (!begrunnelseInneholderFritekst) {
          return harVilkår && valgtBegrunnelseForVilkår;
        }

        return harVilkår && harStrengInnhold(valgtBegrunnelseForVilkår?.begrunnelseFritekst ?? "");
```

Vurderte også å skrive det slik, men det er lettere å lese den jeg comitta selv om det er _mer kode_ 
(Hvis du er veldig uenig om bruk av variabler slik som jeg gjorde, så kan vi selvsagt bruke denne :upside_down_face: )

Reply:
Kommenterte på feil kode, se bort fra denne og se på den andre commenten min som jeg tagga :woozy_face: